robot:
  ros__parameters:
    teb_local_planner:
      odom_topic: "odom"
      map_frame: "odom"
      plugin: "teb_local_planner::TebLocalPlannerROS"
      # footprint_model.type: circular
      # footprint_model.radius: 0.7
      footprint_model.type: "two_circles"
      footprint_model.front_offset: 0.0 # for type "two_circles"
      footprint_model.front_radius: 0.7 # for type "two_circles"
      footprint_model.rear_offset: 0.4 # for type "two_circles"
      footprint_model.rear_radius: 0.6 # for type "two_circles"
      min_obstacle_dist: 0.2
      inflation_dist: 0.7
      costmap_converter_plugin: "costmap_converter::CostmapToPolygonsDBSMCCH"
      costmap_converter_spin_thread: True
      costmap_converter_rate: 15
      enable_homotopy_class_planning: True
      enable_multithreading: True
      optimization_verbose: False
      teb_autoresize: True
      min_samples: 3
      max_samples: 50
      max_global_plan_lookahead_dist: 2.5
      visualize_hc_graph: True
      max_vel_x: 0.3
      max_vel_theta: 0.2
      acc_lim_x: 2.5
      acc_lim_theta: 3.2
      allow_init_with_backwards_motion: false
      weight_kinematics_forward_drive: 100.0